### ISSUE TITLE
Create CHOLMOD sparse matrix from std::vectors

### ISSUE THREAD

**Issue by ducbueno** [DrTimothyAldenDavis/SuiteSparse#117]
Hello everyone. 

This question is not really an issue with the package but a programming difficulty I'm having. If here is not the appropriate place to ask this sort of question, directions to an adequate place would be greatly appreciated.

So, to the question. I'm trying to create a CHOLMOD sparse matrix from 3 std::vectors corresponding to the CSC representation of the sparse matrix (column pointers, row indices and non-zero values). I'm able to allocate the sparse matrix without problems, but my issue begins when I try to attribute the values from my std::vectors to the allocated matrix.

Following the code of CHOLMOD functions that return sparse matrices, this is was the first thing I tried to do:

```c++
int *Sp, *Si;
double *Sx;
cholmod_sparse *S;

S = cholmod_allocate_sparse(s_nrows, s_ncols, s_nnz, 0, 1, 0, CHOLMOD_REAL, cc);
Sp = S->p;
Si = S->i;
Sx = S->x;
//...
```

The above code gives me an error telling me that I'm trying to assign a void pointer to an int pointer. So to solve it, I tried doing the following:

```c++
Sp = (int*) S->p;
Si = (int*) S->i;
Sx = (double*) S->x;
//...
``` 

The code above compiles without errors but gives me a segmentation fault.

Basically,  this is where I'm stuck at and I'm really confusing, since in the CHOLMOD functions I mentioned I see lines of code with both my attempts. 

What am I missing here?

Thanks for the attention.

**Comment by DrTimothyAldenDavis** [comment]
The integers can be either 32-bit or 64-bit.  Make sure you're using the
right set of functions.

std::vectors are likely allocated by "new" not "malloc", and mixing memory
managers will cause a segfault.
That's likely the cause of your segfault.  Don't use std::vector, but use
malloc'd arrays instead.

It's very hard to figure out what you're trying to do, in any case, since I
don't see any std::vectors in the code snippet above.

**Comment by ducbueno** [comment]
Hello everyone. 

Just to let you know I solved the problem. I was calling `cholmod_l_start` and trying to work with regular `int` instead of `SuiteSparse_long`. By calling `cholmod_start` the segfault didn't happen. 

And also passing the values from the `std::vector`s to the sparse matrix structure worked with `std::copy`.
